provider/aws/aws.go

```diff
@@ -333,22 +355,41 @@ func (p *AWSProvider) Zones(ctx context.Context) (map[string]*route53.HostedZone
 				}
 			}
 
-			zones[aws.StringValue(zone.Id)] = zone
+			zones[aws.StringValue(zone.Id)] = &profiledZone{
+				profile: profile,
+				zone:    zone,
+			}
 		}
 
 		return true
 	}
 
-	err := p.client.ListHostedZonesPagesWithContext(ctx, &route53.ListHostedZonesInput{}, f)
-	if err != nil {
-		return nil, provider.NewSoftError(fmt.Errorf("failed to list hosted zones: %w", err))
-	}
-	if tagErr != nil {
-		return nil, provider.NewSoftError(fmt.Errorf("failed to list zones tags: %w", tagErr))
+	for p, client := range p.clients {
+		profile = p
+		err := client.ListHostedZonesPagesWithContext(ctx, &route53.ListHostedZonesInput{}, f)
+		if err != nil {
+			var awsErr awserr.Error
+			if errors.As(err, &awsErr) {
+				switch awsErr.Code() {
+				case "AccessDenied":
+					log.Warnf("Skipping AWS profile %q due to missing permission: %v", profile, awsErr.Message())
```

Comment:
Why this is only a warning and not fatal?
This does not seem to be a temporary issue, but requires attention so we should stop the process.

Reply:
please see answer on next comment. I think both have same scope, aren't they?

Reply:
Fatal would exit the process. As far as I understand we can not recover from this error, so that's what we should do